### ISSUE TITLE
methods of QtInteractor

### ISSUE THREAD

**Issue by bistek** [pyvista/pyvistaqt#17]
Hello, I'm testing some methods of QtInteractor from the new spin-off **pyvistaqt** library.

BTW, it is more clear to have it in a dedicated library. Thanks!

First I create a plotter widget in a Qt QMainWindow created with Qt Designer:

```
class View3D(QMainWindow, Ui_ThreeDWindow):
    def __init__(self, parent=None, *args, **kwargs):
        super(View3D, self).__init__(parent, *args, **kwargs)
        self.setupUi(self)
        self.plotter = QtInteractor(self.ViewFrame)
        self.ViewFrameLayout.addWidget(self.plotter.interactor)
```

Then I am able to add some actor and everything works very well!

`self.plotter.add_mesh(<vtk_object>, name=<vtk_object_name>)`

However some of the methods inherited from **BasePlotter()** crash the application.

Tested working:
self.plotter.show_axes_all()
self.plotter.hide_axes_all()
self.plotter.add_bounding_box()
self.plotter.show_bounds()
self.plotter.show_grid()
self.plotter.add_floor()
self.plotter.add_scalar_bar()

Tested crashing:
self.plotter.add_orientation_widget()
self.plotter.add_toolbars()
self.plotter.add_menu_bar()
self.plotter.add_legend()
self.plotter.default_camera_tool_bar()

**Comment by GuillaumeFavelier** [comment]
Hi @bistek, thanks for doing this manual testing and I'm surprised by the outcome. I think it could be a good idea to have automatic testing for this then once it is fixed.

Among the methods crashing, [`add_toolbars()`](https://github.com/pyvista/pyvistaqt/blob/master/pyvistaqt/plotting.py#L449) and [`add_menu_bar()`](https://github.com/pyvista/pyvistaqt/blob/master/pyvistaqt/plotting.py#L483) are not inherited from `BasePlotter`. Also AFAIK, [`default_camera_tool_bar`](https://github.com/pyvista/pyvistaqt/blob/master/pyvistaqt/plotting.py#L458) is not a function.

**Comment by akaszynski** [comment]
Looks like we indeed need to add automated testing for this.  Thanks for discovering this!

**Comment by bistek** [comment]
I'm happy to help, if I can. Actually YOU are the one who are doing a great job for the community!

**Comment by dominicjtriana** [comment]
I made another issue but I would add:
self.plotter.disable_eye_dome_lighting() throws errors but doesn't crash.  There are some negative visual artifacts as well.

**Comment by condoratberlin** [comment]
Hi, I goning to embedd pyvistaqt in an gui framework an discovered some similar problems. 
Especially `add_toolbars()` requires a `self.app_window` attribute, were the mainwindow is stored, but the pyvista QtInteractor don't allow to set such a parameter to the init method.

**Comment by GuillaumeFavelier** [comment]
I agree with @condoratberlin in here, the `add_toolbars()` and `add_menu_bar()` functions depends on `app_window` internally. Those should not be declared as `QtInteractor` methods.

I see two ways to deal with this at the moment:

1) Just move those 2 to `BackgroundPlotter` (that's where they are used after all)
2) Find a way to refactor the functions to make them useful to `QtInteractor` (requires Qt gymnastics)

About EDL, this was fixed in https://github.com/pyvista/pyvista/pull/1113

**Comment by GuillaumeFavelier** [comment]
Option 1 was implemented in https://github.com/pyvista/pyvistaqt/pull/78

**Comment by GuillaumeFavelier** [comment]
As of now, is there any confusion about the methods of `QtInteractor`?
